src/main/resources/api/service_openapi.yaml

```diff
@@ -216,12 +263,90 @@ components:
             format: uuid
         jobControl:
           $ref: '#/components/schemas/JobControl'
+    CreateDataReferenceRequestBody:
+      type: object
+      properties:
+        name:
+          description: The name of the data reference
+          type: string
+        resourceId:
+          description: The ID of the resource
+          type: string
+          nullable: true
+        referenceType:
+          description: The type of the data reference (i.e. TDR snapshot, AWS bucket, etc)
+          type: string
+          nullable: true
+        reference:
+          description: The reference to the data (i.e. gs://...) #todo- describe as json instead
+          type: object
```

Comment:
Since we're allowing arbitrary json here to make reference types extendible, can this just be a string?

Reply:
If we *expect* json, let's leave it as an `object`.  Does this really allow any string, whether or not it is json?

Reply:
i don't think any string will be valid- it will only ever be json.